File: merge_data.py
import json
import os

data_dir = 'data'

def merge_files(base_file, extra_file):
    base_path = os.path.join(data_dir, base_file)
    extra_path = os.path.join(data_dir, extra_file)
    
    if os.path.exists(base_path) and os.path.exists(extra_path):
        print(f"Merging {base_file} and {extra_file}")
        try:
            with open(base_path, 'r', encoding='utf-8') as f:
                base_data = json.load(f)
            with open(extra_path, 'r', encoding='utf-8') as f:
                extra_data = json.load(f)
                
            if isinstance(base_data, list) and isinstance(extra_data, list):
                existing_ids = {item.get('id') for item in base_data if 'id' in item}
                for item in extra_data:
                    if item.get('id') not in existing_ids:
                        base_data.append(item)
                        existing_ids.add(item.get('id'))
                    else:
                        print(f"- Duplicate ID skipped: {item.get('id')}")
                        
                with open(base_path, 'w', encoding='utf-8') as f:
                    json.dump(base_data, f, indent=2, ensure_ascii=False)
                print(f"- Wrote {len(base_data)} items to {base_file}")
                
                # Cleanup extra file
                os.remove(extra_path)
            else:
                print(f"Skipping {base_file} - content is not a list")
                
        except Exception as e:
            print(f"Error merging {base_file}: {e}")

merge_files('estilos.json', 'estilos-extra.json')
merge_files('interiores.json', 'interiores-extra.json')
merge_files('materiales.json', 'materiales-extra.json')

print("Merge completed.")

File: merge_data.js
const fs = require('fs');
const path = require('path');

const dataDir = path.join(__dirname, 'data');

const mergeFiles = (baseFile, extraFile) => {
    const basePath = path.join(dataDir, baseFile);
    const extraPath = path.join(dataDir, extraFile);
    
    if (fs.existsSync(basePath) && fs.existsSync(extraPath)) {
        console.log(`Merging ${baseFile} and ${extraFile}`);
        
        // Handle direct arrays or object arrays based on file structure
        let baseData = JSON.parse(fs.readFileSync(basePath, 'utf8'));
        let extraData = JSON.parse(fs.readFileSync(extraPath, 'utf8'));
        
        // If they are both arrays
        if (Array.isArray(baseData) && Array.isArray(extraData)) {
            // merge and remove duplicates by ID
            const combined = [...baseData];
            const existingIds = new Set(baseData.map(item => item.id));
            
            for (const item of extraData) {
                if (!existingIds.has(item.id)) {
                    combined.push(item);
                    existingIds.add(item.id);
                } else {
                    console.log(`- Duplicate ID found/skipped: ${item.id}`);
                }
            }
            fs.writeFileSync(basePath, JSON.stringify(combined, null, 2), 'utf8');
            console.log(`- Wrote ${combined.length} items to ${baseFile}`);
            
            // Delete extra files after merge to clean up
            fs.unlinkSync(extraPath);
        } else {
            console.log(`Could not merge ${baseFile} (Not both arrays)`);
        }
    }
};

mergeFiles('estilos.json', 'estilos-extra.json');
mergeFiles('interiores.json', 'interiores-extra.json');
mergeFiles('materiales.json', 'materiales-extra.json');

console.log("Merge completed.");
